Source/Core/VideoCommon/FrameDump.cpp

```diff
@@ -336,10 +336,9 @@ void FrameDump::ProcessPackets()
 {
   while (true)
   {
-    AVPacket pkt;
-    av_init_packet(&pkt);
+    AVPacket* pkt = av_packet_alloc();
 
-    const int receive_error = avcodec_receive_packet(m_context->codec, &pkt);
+    const int receive_error = avcodec_receive_packet(m_context->codec, pkt);
 
     if (receive_error == AVERROR(EAGAIN) || receive_error == AVERROR_EOF)
     {
```

Comment:
This statement causes `pkt` to leave scope without being freed, same with any other statement that breaks this loop. I suggest wrapping `pkt` in a smart pointer.

Reply:
Indeed, that’s now fixed with a `std::unique_ptr` with a deleter.

Reply:
As a less verbose alternative to unique_ptr's deleter syntax:
```
  AVPacket* pkt = av_packet_alloc();
  Common::ScopeGuard pkt_guard{[&pkt]() { av_packet_free(&pkt); }};
```
... which also frees you from having to call get() everywhere. (ScopeGuard is in Common/ScopeGuard.h)